api/graphql/types/allocated_time_slot/serializers.py

```diff
@@ -0,0 +1,237 @@
+from rest_framework import serializers
+from rest_framework.settings import api_settings
+
+from applications.choices import Weekday
+from applications.models import AllocatedTimeSlot, ApplicationSection, ReservationUnitOption, SuitableTimeRange
+from common.date_utils import time_difference, timedelta_to_json
+
+__all__ = [
+    "AllocatedTimeSlotCreateSerializer",
+]
+
+
+class AllocatedTimeSlotCreateSerializer(NestingModelSerializer):
+    force = serializers.BooleanField(default=False, required=False, write_only=True)
+
+    class Meta:
+        model = AllocatedTimeSlot
+        fields = [
+            "pk",
+            "day_of_the_week",
+            "begin_time",
+            "end_time",
+            "reservation_unit_option",
+            "force",
+        ]
+
+    def validate(self, data: dict[str, Any]) -> dict[str, Any]:
+        errors: dict[str, list[str]] = defaultdict(list)
+        force: bool = data.pop("force", False)
+
+        section: ApplicationSection = data["reservation_unit_option"].application_section
+
+        self.validate_statuses(
+            section=section,
+            errors=errors,
+        )
+        self.validate_reservation_unit_option_available(
+            option=data["reservation_unit_option"],
+            errors=errors,
+        )
+        self.validate_overlapping_allocations(
+            section=section,
+            day_of_the_week=data["day_of_the_week"],
+            begin_time=data["begin_time"],
+            end_time=data["end_time"],
+            option=data["reservation_unit_option"],
+            errors=errors,
+        )
+        self.validate_no_allocations_on_the_same_day(
+            section=section,
+            day_of_the_week=data["day_of_the_week"],
+            errors=errors,
+        )
+        self.validate_slots_per_week(
+            section=section,
+            errors=errors,
+        )
+
+        if not force:
+            self.validate_duration(
+                section=section,
+                begin_time=data["begin_time"],
+                end_time=data["end_time"],
+                errors=errors,
+            )
+            self.validate_day_of_the_week_is_suitable(
+                section=section,
+                day_of_the_week=data["day_of_the_week"],
+                errors=errors,
+            )
+            self.validate_within_suitable_time_range(
+                section=section,
+                day_of_the_week=data["day_of_the_week"],
+                begin_time=data["begin_time"],
+                end_time=data["end_time"],
+                errors=errors,
+            )
+
+        if errors:
+            raise serializers.ValidationError(errors)
+
+        return data
+
+    def validate_statuses(
+        self,
+        section: ApplicationSection,
+        errors: dict[str, list[str]],
+    ) -> None:
+        section_status = section.status
+        application_status = section.application.status
+
+        if not section_status.can_approve:
```

Comment:
I guess section_status.can_approve could be renamed can_allocate

Reply:
Done.